toolset/benchmark/framework_test.py

```diff
@@ -278,18 +212,24 @@ def tee_output(prefix, line):
               tee_output(prefix, line)
           except setup_util.EndOfStream:
             break
-      p = subprocess.Popen(["docker", "build", "-f", test_docker_file, "-t", "tfb-test-%s" % self.name, self.directory],
-          stdout=subprocess.PIPE,
-          stderr=subprocess.STDOUT)
-      nbsr = setup_util.NonBlockingStreamReader(p.stdout)
-      while (p.poll() is None):
-        for i in xrange(10):
-          try:
-            line = nbsr.readline(0.05)
-            if line:
-              tee_output(prefix, line)
-          except setup_util.EndOfStream:
-            break
```

Comment:
All the above was done inside the loop that builds the dependencies... this should NEVER have worked and yet we got here somehow. 😕

Reply:
I get it - it tries to build the test's dockerfile every time in the loop, and we never checked the success of this until this PR where it returns and exits upon failure. It worked before because it would fail every time until all the dependencies had successfully built. 🤣